frontend/src/Components/Basket.js

```diff
@@ -4,57 +4,64 @@ import { useNavigate } from "react-router-dom";
 import ProductList from "./ProductList";
 import BasketItem from "./BasketItem";
 import ShopService from "../Services/ShopService";
+import PaymentForm from "./PaymentForm";
 
 const DELIVERY_FEE = 1;
 
-export default function Basket() {
+export default function Basket(props) {
+  let orderData = JSON.stringify(props.order);
```

Comment:
This is never reassigned so can be a `const`

```suggestion
  const orderData = JSON.stringify(props.order);
```